state/src/behavior.ts

```diff
@@ -0,0 +1,181 @@
+import {EditorState, StateField} from "./state"
+
+type A<T> = ReadonlyArray<T>
+
+export enum Priority { fallback = -1, base = 0, extend = 1, override = 2 }
```

Comment:
If you don't want this to appear in compiled code, check out [const enums](https://www.typescriptlang.org/docs/handbook/enums.html#const-enums) (I think I mentioned them once).

Reply:
These have to be available to client code, which may not be using TypeScript, so I think we have to use a dynamic 'enum' here (TypeScript appears to compile const enums to nothing at all).